test/fixtures/rules/std_rule_cases/L003.yml

```diff
@@ -620,6 +663,44 @@ test_tsql_function:
     core:
       dialect: tsql
 
+test_stable_jinja_loop:
+  fail_str: |
+    {% for t in ['table_a', 'table_b'] %}
+        SELECT
+            column_a,
+            column_b
+        FROM {{ t }}
+            {% if not loop.last %}
+                UNION ALL
+            {% endif %}
+    {% endfor %}
+    ;
+  fix_str: |
+    {% for t in ['table_a', 'table_b'] %}
+        SELECT
+            column_a,
+            column_b
+        FROM {{ t }}
+        {% if not loop.last %}
+            UNION ALL
+        {% endif %}
+    {% endfor %}
+    ;
+
+test_stable_jinja_loop:
+  # Matches the fix output from above
```

Comment:
IIUC, the YAML test framework nowadays checks to confirm that the `fix_str` of a test yields no subsequent fixes. If so, this test case may not be needed. Can you confirm this, @tunetheweb?

Reply:
Not sure that could be true. One of the issues with L003 is that it was removing indent markers sometimes. causing the next pass to actually have different markers. It would be stable for a single run (as the parse tree would reach something.)

Reply:
Maybe I was thinking of [this code](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/testing/rules.py#L210L214): Tests _without_ a `fix_str` do not apply any fixes.

We should probably modify the test framework to detect and fail if a second round of fixing makes additional changes.